File: segundos 4 lenguajes/c#/ProyectoCsharp/JugadorCurador.cs
namespace ProyectoJugadores
{
    public class JugadorCurador
    {
        public string Nombre { get; set; }
        public int Nivel { get; set; }

        public JugadorCurador()
        {
            Nombre = "";
            Nivel = 0;
        }

        public JugadorCurador(string nombre, int nivel)
        {
            Nombre = nombre;
            Nivel = nivel;
        }

        public static JugadorCurador Copy(JugadorCurador j)
        {
            return new JugadorCurador(j.Nombre, j.Nivel);
        }
    }
}

File: segundos 4 lenguajes/go/JugadorCurador.go
package main

type JugadorCurador struct {
	Nombre string
	Nivel  int
}

func NewJugadorCurador() JugadorCurador {
	return JugadorCurador{"", 0}
}

func NewJugadorCuradorParams(nombre string, nivel int) JugadorCurador {
	return JugadorCurador{nombre, nivel}
}

func NewJugadorCuradorNombre(nombre string) JugadorCurador {
	return JugadorCurador{nombre, 1}
}

func NewJugadorCuradorCopy(j JugadorCurador) JugadorCurador {
	return JugadorCurador{j.Nombre, j.Nivel}
}

type AbstractCurador interface {
	Curar()
}
